corehq/apps/users/tasks.py

```diff
@@ -376,6 +361,31 @@ def process_reporting_metadata_staging():
         process_reporting_metadata_staging.delay()
 
 
+def _process_reporting_metadata_staging(chunk_size=100):
+    from corehq.apps.users.models import UserReportingMetadataStaging
+    for pk in UserReportingMetadataStaging.objects.values_list('pk', flat=True).order_by('pk')[:chunk_size]:
+        with transaction.atomic():
+            record = UserReportingMetadataStaging.objects.select_for_update(skip_locked=True).get(pk=pk)
+            try:
+                _process_record_with_retry(record)
+            except Exception:
+                notify_exception(None, "Error encountered in _process_reporting_metadata_staging")
```

Comment:
Would it be helpful to know which record failed, in case it is a persistent failure that keeps happening over and over? Or is it better to lump them all into a single Sentry error group with the details (like which record failed) available in the stack trace introspection?

Alternate error message, unique per failed record
```suggestion
                notify_exception(None, f"Error processing UserReportingMetadataStaging pk={pk}")
```